src/main.c

```diff
@@ -41,11 +41,40 @@ main(int argc, char** argv)
 {
   hiwire_setup();
 
-  setenv("PYTHONDONTWRITEBYTECODE", "1", 0);
   setenv("PYTHONHOME", "/", 0);
 
   Py_InitializeEx(0);
 
+  // This doesn't seem to work anymore, but I'm keeping it for good measure anyway
+  // The effective way to turn this off is below: setting sys.done_write_bytecode = True
+  setenv("PYTHONDONTWRITEBYTECODE", "1", 0);
+
+  PyObject* sys = PyImport_ImportModule("sys");
+  if (sys == NULL) {
+    return 1;
+  }
+  if (PyObject_SetAttrString(sys, "dont_write_bytecode", Py_True)) {
+    Py_DECREF(sys);
```

Comment:
Is there any reason to decrement the reference count if we're just going to return 1 anyways?

Reply:
Probably not from the main function here, no.  Good point.